ref-exchange/src/lib.rs

```diff
@@ -149,16 +150,11 @@ impl Contract {
     /// If there is not enough attached balance to cover storage, fails.
     /// If too much attached - refunds it back.
     fn internal_add_pool(&mut self, pool: Pool) -> u64 {
-        let prev_storage = env::storage_usage();
+        let start_storage = env::storage_usage();
         let id = self.pools.len() as u64;
         self.pools.push(&pool);
 
-        // Check how much storage cost and refund the left over back.
-        let storage_cost = (env::storage_usage() - prev_storage) as u128 * env::storage_byte_cost();
-        assert!(
-            storage_cost <= env::attached_deposit(),
-            "ERR_STORAGE_DEPOSIT"
-        );
+        // TODO: update deposit handling
         let refund = env::attached_deposit() - storage_cost;
         if refund > 0 {
             Promise::new(env::predecessor_account_id()).transfer(refund);
```

Comment:
refund won't work, because I added `assert_one_yocto()`.

Reply:
Moreover - it's easier to standarize here to use internal storage tracking + `assert_one_yocto`, rather than asking user for unknown required NEAR attachment.